processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,9 +82,55 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
+	clusterName, err := d.getClusterName()
+	if err != nil {
+		return pcommon.NewResource(), "", err
```

Comment:
Will this be breaking for users that don't have the correct permissions?  Would returning `d.rb.Emit(), conventionss.SchemaURL, err` be better, or would the processor still use an empty resource in that case?

Reply:
Good call. It's better to return as much as possible, especially when the other values here aren't dependent on the cluster name (and we've already successfully determined it's an EKS cluster). I'll update.